src/hip_memory.cpp

```diff
@@ -1278,8 +1282,12 @@ hipError_t hipMemcpyFromSymbol(void* dst, const void* src, size_t count,
 
     tprintf(DB_MEM, " symbol '%s' resolved to address:%p\n", symbol_name, dst);
 
-    return ihipLogStatus(
-        hipMemcpy(dst, static_cast<const char*>(src) + offset, count, kind));
+    if (src == nullptr || dst == nullptr) {
```

Comment:
Since you are adding error checks don't you think symbol_name and hipMemcpyKind should also be validated?

Reply:
symbol_name is validated as part of inline implementation @ include/hip/hcc_detail/hip_runtime_api.h:1594. hipMemcpyKind is used as part of a switch case which defaults to generic copy if the kind is invalid